client/my-sites/themes/validate-filters.js

```diff
@@ -30,5 +31,21 @@ module.exports = function validateFilter( context, next ) {
 	}
 
 	next();
-};
+}
 
+export function validateVertical( context, next ) {
+	const { vertical } = context.params;
+
+	if ( ! vertical ) {
+		return next();
+	}
+
+	if ( ! includes( getSubjects(), vertical ) ) {
+		if ( context.isServerSide ) {
+			return context.res.redirect( '/themes' );
+		}
+		return page.redirect( '/themes' );
```

Comment:
Turns out the reconc error pre-exists (happens on `master` for `/themes/blergh`), so I'll try and fix that separately. So the last diff should be good, tho I think we should just bail on the client route:

```diff
diff --git a/client/my-sites/themes/validate-filters.js b/client/my-sites/themes/validate-filters.js
index 77f33f4..7d9cfbe 100644
--- a/client/my-sites/themes/validate-filters.js
+++ b/client/my-sites/themes/validate-filters.js
@@ -42,9 +42,10 @@ export function validateVertical( context, next ) {
 
        if ( ! includes( getSubjects(), vertical ) ) {
                if ( context.isServerSide ) {
-                       return context.res.redirect( '/themes' );
+                       return next( 'route' );
                }
-               return page.redirect( '/themes' );
+               // Client-side: Terminate routing, rely on server-side rendered markup.
+               return;
        }
 
        next();
```